lib/Doctrine/ORM/Mapping/DefaultTypedFieldMapper.php

```diff
@@ -36,12 +57,47 @@ final class DefaultTypedFieldMapper implements TypedFieldMapper
         'string' => Types::STRING,
     ];
 
+    /** It maps built-in Doctrine types to PHP types */
+    private const BUILTIN_TYPES_MAP = [
+        AsciiStringType::class => 'string',
+        BigIntType::class => 'string',
+        BooleanType::class => 'bool',
+        DateType::class => DateTime::class,
+        DateImmutableType::class => DateTimeImmutable::class,
+        DateIntervalType::class => DateInterval::class,
+        DateTimeType::class => DateTime::class,
+        DateTimeImmutableType::class => DateTimeImmutable::class,
+        DateTimeTzType::class => DateTime::class,
+        DateTimeTzImmutableType::class => DateTimeImmutable::class,
+        DecimalType::class => 'string',
+        FloatType::class => 'float',
+        GuidType::class => 'string',
+        IntegerType::class => 'int',
+        JsonType::class => 'array',
+        SimpleArrayType::class => 'array',
+        SmallIntType::class => 'int',
+        StringType::class => 'string',
+        TextType::class => 'string',
+        TimeType::class => DateTime::class,
+        TimeImmutableType::class => DateTimeImmutable::class,
+    ];
```

Comment:
Discussed this with @DavideBicego in a preparation PR: the idea here is that the **exact** DBAL type must be used (no subclasses), since userland consumers of `doctrine/orm` may have their own customization around field types.

It is worth documenting such a detail in the docblock.

Reply:
I think this is a detail for which we should write a small regression test too, to prevent misuse in future refactoring attempts.